File: src/main/java/com/adinsight/agent/loop/AgentLoopResult.java
package com.adinsight.agent.loop;

import java.util.Map;

public record AgentLoopResult(
        String content,
        String model,
        Map<String, Object> usage,
        FinishReason finishReason,
        int turnCount
) {
    public AgentLoopResult {
        usage = usage == null ? Map.of() : Map.copyOf(usage);
    }
}

File: src/loop/agent-loop-result.ts
import type { FinishReason } from "./finish-reason.js";

export type Usage = Readonly<Record<string, unknown>>;

export type AgentLoopResult = Readonly<{
  content: string;
  model?: string;
  usage: Usage;
  finishReason: FinishReason;
  turnCount: number;
}>;
